cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/service/http/BasicHttpServiceContext.java

```diff
@@ -106,13 +106,13 @@ public TransactionContext getTransactionContext() {
   }
 
   private static MetricsCollector getMetricCollector(MetricsCollectionService service,
-                                                     Program program, String runnableName,
+                                                     Program program, String handlerName,
                                                      String runId, int instanceId) {
     if (service == null) {
       return null;
     }
     Map<String, String> tags = Maps.newHashMap(getMetricsContext(program, runId));
-    tags.put(Constants.Metrics.Tag.SERVICE_RUNNABLE, runnableName);
+    tags.put(Constants.Metrics.Tag.HANDLER, handlerName);
```

Comment:
question : do we care about the old data that's aggregated under SERVICE_RUNNABLE ?